### PR TITLE
PandasTypeSelector for selecting pandas columns based on dtypes

### PR DESCRIPTION
**Pull request by kayhoogland**
Added PandasTypeSelector to select columns in a pandas dataframe based on their dtype.

### CODE REVIEW

File: sklego/preprocessing.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+
+class PandasTypeSelector(BaseEstimator, TransformerMixin):
+    """
+    Select columns in a pandas dataframe based on their dtype
+
+    :param include: types to be included in the dataframe
+    :param exclude: types to be exluded in the dataframe
+    """
+    def __init__(self, include, exclude=[]):
```
Comment (koaning) [inline sklego/preprocessing.py]: could you turn the empty list into something that is not mutable? 

also, should `include` also not be optional?

File: sklego/preprocessing.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        self.exclude = exclude
+
+    def fit(self, X, y=None):
+        """
+        Saves the column names for check during transform
+
+        :param X: pandas dataframe to select dtypes for
+        :param y: not used in this class
+        """
+        self._check_X_for_type(X)
```
Comment (koaning) [inline sklego/preprocessing.py]: maybe an explicit `ValueError` when you select columns that do not appear in the dataframe that is brought in here? 

you probably also want to do this during the `.transform()` phase

Comment (kayhoogland) [reply sklego/preprocessing.py]: Yes I changed it

File: sklego/preprocessing.py (hunk 1 above)
Comment (kayhoogland) [reply sklego/preprocessing.py]: Good point, changed the defaults to None

File: sklego/preprocessing.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    """
+    def __init__(self, include=None, exclude=None):
+        self.include = include
+        self.exclude = exclude
+
+    def fit(self, X, y=None):
+        """
+        Saves the column names for check during transform
+
+        :param X: pandas dataframe to select dtypes for
```
Comment (koaning) [inline sklego/preprocessing.py]: for -> out of

File: tests/test_transformers/test_pandastypeselector.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+    with pytest.raises(ValueError):
+        transformer.fit(X, y)
+
+
+def test_value_error_inequal(random_xy_dataset_regr):
+    transformer = PandasTypeSelector(include=['number'])
+    X, y = random_xy_dataset_regr
+    X = pd.DataFrame(X)
+    if len(list(X)) > 0:
```
Comment (koaning) [inline tests/test_transformers/test_pandastypeselector.py]: use `X.shape[0]` -> this is always more performant and direct.

File: sklego/preprocessing.py (hunk 3 above)
Comment (kayhoogland) [reply sklego/preprocessing.py]: Check, changed

File: tests/test_transformers/test_pandastypeselector.py (hunk 4 above)
Comment (kayhoogland) [reply tests/test_transformers/test_pandastypeselector.py]: Agreed, changed in last commit

**Comment by kayhoogland** [comment]
https://github.com/koaning/scikit-lego/issues/12

**Pull request merged**